Add federate resource quota docs (#249)
karmada-io/website: docs/userguide/roleseparation/federated-resource-quota.md

Resource administrators often need to stand on the global dimension to manage and control the total consumption of resources by each business.

Creating a corresponding `namespace` and `ResourceQuotas` under each Kubernetes cluster is the usual practice, and then Kubernetes will limit the resources by `ResourceQuotas`. 
However, with the growth of the number of businesses, the expansion and contraction of member clusters, the number of available resources and resource types of each cluster are different, which brings many problems to the administrator.

In addition, Karmada supports the propagation of ResourceQuota objects to Kubernetes clusters through PropagationPolicy, so as to achieve the purpose of creating quotas on multi-clusters by kubernetes native APIs. 
However, it is impossible to freely adjust and limit the global resource usage of a business by PropagationPolicy, and it is quite troublesome to expand such a requirement on it. 
We need a global quota for Karmada, not just ResourceQuota on member clusters.

> The `FederatedResourceQuota` API is provided to solve the above problems. Here we give several typical usage scenarios of `FederatedResourceQuota`.

Reviewer: ```suggestion
The `FederatedResourceQuota` API is provided to solve the above problems. Here we give several typical usage scenarios of `FederatedResourceQuota`.
```
```suggestion
Here's where the `FederatedResourceQuota` API comes in. The following is several typical usage scenarios of `FederatedResourceQuota`.
```

## What can FederatedResourceQuota do

FederatedResourceQuota supports:

> * The global quota management for multiple clusters managed by Karmada.

Reviewer: ```suggestion
* Global quota management for multiple clusters managed by Karmada.
```

* Fine-grained management of quotas under the same namespace of different clusters.
    * Ability to enumerate resource usage limits per namespace.
    * Ability to monitor resource usage for tracked resources.
    * Ability to reject resource usage exceeding hard quotas.

![unified resourcequota](../../resources/key-features/unified-resourcequota.png)

> FederatedResourceQuota supports the management of the following resources: cpu, memory, storage and ephemeral-storage.

Reviewer: ```suggestion
You can use FederatedResourceQuota to manage CPU, memory, and storage (including the ephemeral one).
```

## Deploy a simplest FederatedResourceQuota

> For example, as an administrator, you want to deploy business A across multiple clusters under namespace `test`.

Reviewer: ```suggestion
Assume you, an administrator, want to deploy service A across multiple clusters in namespace `test`.
```

You can create a namespace called test on the Karmada control plane. Karmada will automatically create the corresponding namespace in the member clusters.

```shell
kubectl --kubeconfig ~/.kube/karmada.config --context karmada-apiserver create ns test
```

> You want to create a resource limit of 100 core CPUs for business A.

Reviewer: ```suggestion
You want to create a CPU limit of 100 cores for service A.
```

The available resources on clusters:

* member1: 20C
* member2: 50C

> * member3: 100C

Reviewer 2: This number seems incorrect.

Author: The numbers here are allocatable resources.

> In order to better utilize cluster resources, the administrator want to allocate 20C of member1, 50C of member2 and 30C of member3 for business A. 

Reviewer: ```suggestion
In order to better utilize cluster resources, the administrator want to allocate 20C of member1, 50C of member2 and 30C of member3 for business A. 
```
```suggestion
In this example, allocate 20C of member1, 50C of member2, and 30C of member3 to service A. 
```

> You want to reserve resources for more important business in member3.

Reviewer: ```suggestion
Resources are reserved for more important services in member3.
```

    hard:
      cpu: 20
  - clusterName: member2
    hard:
      cpu: 50
  - clusterName: member3
    hard:
      cpu: 30
status:
  aggregatedStatus:
  - clusterName: member1
    hard:
      cpu: "20"
    used:
      cpu: "0"
  - clusterName: member2
    hard:
      cpu: "50"
    used:
      cpu: "0"
  - clusterName: member3
    hard:
      cpu: "30"
    used:
      cpu: "0"
  overall:
    cpu: "100"
  overallUsed:
    cpu: "0"
```

For quick test, you can deploy a simple application which consumes 1C CPU to member1.

```shell
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
  namespace: test
  labels:
    app: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - image: nginx
        name: nginx
        resources:
          requests:
             cpu: 1
```

Verify the status of FederatedResourceQuota and you will find that FederatedResourceQuota can monitor resource usage correctly for tracked resources.

```shell
kubectl --kubeconfig ~/.kube/karmada.config --context karmada-apiserver get federatedresourcequotas/test -ntest -oyaml
```

```
spec:

>   overall:

Reviewer: Not sure this is fixed or changeable. IMO, "total" is a better word.

---
title: Federated ResourceQuota
---

## Background

With the widespread use of multi-clusters, the administrator may deploy services on multiple clusters and resource management of services under multiple clusters has become a new challenge.
A traditional approach is that the administrator manually deploys  a `namespace` and `ResourceQuota` under each Kubernetes cluster, where Kubernetes will limit the resources according to the `ResourceQuotas`.

> This approach is a bit cumbersome and not flexible enough.

Reviewer 3: Do we have any alternative for `cumbersome`? I'd prefer the one common and simple.
@Tingtal

Reviewer: `cumbersome` is acceptable to me. Another way to say it could be:
The disadvantage is that you need to do it one by one, not once for all.

Author: Change to `inconvenient`

In addition, this practice is now challenged by the differences in service scale, available resources and resource types of each cluster.

Resource administrators often need to manage and control the consumption of resources by each service in a global view.
Here's where the `FederatedResourceQuota` API comes in. The following is several typical usage scenarios of `FederatedResourceQuota`.

## What FederatedResourceQuota can do

FederatedResourceQuota supports:

> * Global quota management for multiple clusters managed by Karmada.

Reviewer 3: ```suggestion
* Global quota management for applications that run on multiple clusters.
```
The quota feature actually manages applications, not clusters.
